semgrep/semgrep/rule_lang.py

```diff
@@ -185,10 +186,16 @@ def unroll(self) -> YamlValue:
             return {str(k.unroll()): v.unroll() for k, v in self.value.items()}
         elif isinstance(self.value, YamlTree):
             return self.value.unroll()
-        elif isinstance(self.value, str) or isinstance(self.value, int):
+        elif (
+            isinstance(self.value, str)
+            or isinstance(self.value, int)
+            or self.value is None
```

Comment:
Is this so we can handle the `None` case later? Can we handle it here rather than pass the value along and deal with it later?

Reply:
This is actually the unrolling process (where we convert YAML into a regular dict to be used in cases where we don't care about spans anymore). 

By handling the check in the YamlMap constructor, we're handling it as early as possible (at YAML parse time rather than failing later during unrolling).

This check could _probably_ actually be deleted given the other code, but I figured it was probably better not to explode with a strange error as we did to Clint. 

It's also not 100% clear to me we never want to allow explicit nulls, eg. in a list.

Reply:
Ahh, interesting, good catch.

> This check could probably actually be deleted given the other code, but I figured it was probably better not to explode with a strange error as we did to Clint.

I would actually argue that this is beneficial. Failing loudly and handling that case will let us know vs. failing silently with unexpected behavior somewhere later in the program.

> It's also not 100% clear to me we never want to allow explicit nulls, eg. in a list.

I think this is what we _currently_ want - pretty much everything is a string at the moment. Again, I'd rather fail loudly and handle nulls if/when we need them in the future vs. trying to accommodate future functionality. 

Thoughts?